tests/gtest/are_images_equal.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+// Compares two files and returns whether they are the same once decoded.
+
+#include <iostream>
+
+#include "aviftest_helpers.h"
+#include "avifutil.h"
+
+using libavif::testutil::AvifImagePtr;
+
+int main(int argc, char** argv) {
+  if (argc != 4) {
+    std::cerr
+        << "Wrong argument: ./are_image_equal file1 file2 ignore_alpha_flag"
+        << std::endl;
+    return 1;
+  }
+  AvifImagePtr decoded[2] = {
+      AvifImagePtr(avifImageCreateEmpty(), avifImageDestroy),
+      AvifImagePtr(avifImageCreateEmpty(), avifImageDestroy)};
+  if (!decoded[0] || !decoded[1]) {
+    std::cerr << "Cannot create AVIF images." << std::endl;
+    return 2;
+  }
+  uint32_t depth[2];
+  constexpr int kRequestedDepth = 8;
```

Comment:
It seems that we should set kRequestedDepth to 0. 0 means "request the bit depth closest to the bit depth of the input file".

The `avifImage` cannot use the exact bit depth of the input fie when the input file is 16-bit PNG; we are forced to use 12-bit AVIF in that case.

For lossless test, if the bit depth of the `avifImage` is not equal to the bit depth of the input file, there is potentially loss from the `avifReadImage()` call.